### PR TITLE
Mission brief for Thrice

### PR DESCRIPTION
**Pull request by kxmbrian**

### CODE REVIEW

File: briefs/m5-thrice.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+8.0
+>>> (((max)))(1,9,3)
+9
+>>> (add1)(0)
+1
+```
+
+*"In that case, how do I tell when the parentheses indicate a function call and when I can ignore them? :confused:"*
+
+One heuristic is to look at what precedes the open parenthesis `(`:
```
Comment (fonglh) [inline briefs/m5-thrice.md:86]: Might be clearer to put quote marks around the ( to show we're referring to the character? But then it won't look like a smiley, haha. your call.

File: briefs/m5-thrice.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+= (lambda x: add2(add4(x)))(0)
+=            add2(add4(0))
+=            add2(4      )
+= 6
+```
+
+
+### Expression 2
+
+We commence the trace
```
Comment (fonglh) [inline briefs/m5-thrice.md]: This sounds a bit aloof. Just remove this line?

File: briefs/m5-thrice.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+```python
+>>> thrice(double(add1))(0)  # Expression 1
+6
+>>> thrice(double)(add1)(0)  # Expression 2
+8
+```
+
+How cool is that? Same same but different hor.
+
+This article is just to aid your understanding. For sanity, please do not write out all the steps in your mission solution.
```
Comment (fonglh) [inline briefs/m5-thrice.md]: > For your own sanity (and ours), please do not ...

**Comment by fonglh** [comment]
Just a few very minor wording suggestions.

This is a very clear explanation :+1:

**Comment by fonglh** [comment]
Need to add an entry to summary.md too so it'll show up.

File: briefs/m5-thrice.md (hunk 1 above)
Comment (kxmbrian) [inline briefs/m5-thrice.md:86]: Tried that. I think it adds to the clutter and the code font is enough

File: briefs/m5-thrice.md (hunk 2 above)
Comment (kxmbrian) [inline briefs/m5-thrice.md]: okay thanks

File: briefs/m5-thrice.md (hunk 3 above)
Comment (kxmbrian) [inline briefs/m5-thrice.md]: ok, I'll add this for our own sanity (and theirs). =p

**Comment by kxmbrian** [comment]
@fonglh Fixed. Thanks for reviewing!

File: SUMMARY.md [hunk 4]
```diff
@@ -9,3 +9,4 @@
 * [Orders of Growth](chapters/4-complexity.md)
 * Mission Briefs
    * [M2: Egyptian](briefs/m2-egyptian.md)
+   * [M2: More than Thrice](briefs/m5-thrice.md)
```
Comment (fonglh) [inline SUMMARY.md]: Description text should be M5.

Comment (kxmbrian) [inline SUMMARY.md]: whoops. fixed.

**Pull request merged**